packages/eas-cli/src/build/utils/repository.ts

```diff
@@ -9,51 +8,12 @@ import { v4 as uuidv4 } from 'uuid';
 import Log from '../../log';
 import { confirmAsync, promptAsync } from '../../prompts';
 import { formatBytes } from '../../utils/files';
-import {
-  doesGitRepoExistAsync,
-  getGitDiffOutputAsync,
-  gitDiffAsync,
-  gitRootDirectoryAsync,
-  gitStatusAsync,
-  isGitInstalledAsync,
-} from '../../utils/git';
 import { getTmpDirectory } from '../../utils/paths';
 import { endTimer, formatMilliseconds, startTimer } from '../../utils/timer';
+import vcs from '../../vcs';
 
-async function ensureGitRepoExistsAsync(): Promise<void> {
-  if (!(await isGitInstalledAsync())) {
-    throw new Error('git command not found, install it before proceeding');
-  }
-
-  if (await doesGitRepoExistAsync()) {
-    return;
-  }
-
-  Log.warn("It looks like you haven't initialized the git repository yet.");
-  Log.warn('EAS Build requires you to use a git repository for your project.');
-
-  const confirmInit = await confirmAsync({
-    message: `Would you like to run 'git init' in the current directory?`,
-  });
-  if (!confirmInit) {
-    throw new Error(
-      'A git repository is required for building your project. Initialize it and run this command again.'
-    );
-  }
-  await spawnAsync('git', ['init']);
-
-  Log.log("We're going to make an initial commit for you repository.");
-
-  await commitPromptAsync({ initialCommitMessage: 'Initial commit', commitAllFiles: true });
-}
-
-async function isGitStatusCleanAsync(): Promise<boolean> {
-  const changes = await gitStatusAsync({ showUntracked: true });
-  return changes.length === 0;
-}
-
-async function maybeBailOnGitStatusAsync(): Promise<void> {
-  if (await isGitStatusCleanAsync()) {
+export async function maybeBailOnRepoStatusAsync(): Promise<void> {
+  if (await vcs.hasUncommittedChangesAsync()) {
     return;
   }
   Log.addNewLineIfNone();
```

Comment:
I can't comment on the line below (`git working tree`) but please make it more generic.